### DISCUSSION TITLE
How do I use a Solid Store as the data source?

### DISCUSSION THREAD

**Discussion by escritorio-gustavo** [TanStack/table#5041] [Q&A]
I have a table component that invokes `createSolidTable` with the following code:
```tsx
type TableProps<T, U> = Omit<JSX.HTMLAttributes<HTMLTableElement>, 'children'> & {
  data: T[]
  ref?: HTMLTableElement
  columns: ColumnDef<T, U>[]
  pageSize?: number
}

export function Table<T, U>(props: TableProps<T, U>) {
  const [local, rest] = splitProps(props, ['data', 'columns', 'pageSize', 'class'])

  const table = createSolidTable<T>({
    get data() {
      return local.data
    },
    get columns() {
      return local.columns
    },
    getCoreRowModel: getCoreRowModel(),
    getFilteredRowModel: getFilteredRowModel(),
    getPaginationRowModel: getPaginationRowModel(),
  })

  return (
    // render code...
  )
}
```

This works fine, but if I have data which I keep in a Solid store as such:
```tsx
function Foo() {
  const [state, setState] = createStore({ data: [] })

  return <Table data={state.data} columns={/* columnDef */} pageSize={100} />
}
```

This will not update the table when `state.data` is changed through `setState` unless I use `data={[...state.data]}`, which degrades performance a lot and kinda defeats the point of using a store in the first place.

Does anyone know how to make this work without copying the data all the time?

**Comment by AlexErrant** [comment]
I _somewhat_ answer this question here https://github.com/TanStack/table/issues/5019#issuecomment-2634580600
